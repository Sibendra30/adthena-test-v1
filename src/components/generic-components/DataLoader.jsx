import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const BlueDiv = styled.div`
  color: #216298;
  font-weight: bold
`;

// eslint-disable-next-line react/prop-types
function DataLoader({ textValue }) {
  return (
    <div>
      <img src="/images/data-loader.gif" alt="Loading" />
      <BlueDiv>{textValue}</BlueDiv>
    </div>
  );
}

DataLoader.prototype = {
  text: PropTypes.string
};

export default DataLoader;
