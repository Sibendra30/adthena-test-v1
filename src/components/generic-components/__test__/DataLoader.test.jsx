import React from 'react';
import { render } from '@testing-library/react';
import DataLoader from '../DataLoader';

describe('Testing Data Loader Component', () => {
  it('Loader image should render', () => {
    const { container } = render(<DataLoader />);
    expect(container).toMatchSnapshot();
  });

  it('Loader image should be present', () => {
    const { getByAltText } = render(<DataLoader />);
    expect(getByAltText('Loading')).not.toBeNull();
  });
});
