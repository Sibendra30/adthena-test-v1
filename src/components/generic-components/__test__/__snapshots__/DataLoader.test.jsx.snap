// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Data Loader Component Loader image should render 1`] = `
<div>
  <div
    class="data-loader"
  >
    <img
      alt="Loading"
      src="/images/data-loader.gif"
    />
    <div
      class="DataLoader__BlueDiv-sc-15ebnvh-0 gnQMPn"
    />
  </div>
</div>
`;
